Updated ASDF usage to remove warnings

vsedach opened the pull request:
SYSTEM-DEFINITION-PATHNAME is deprecated, replace with
SYSTEM-SOURCE-DIRECTORY

Follow ASDF3 best practices for system definition:
https://github.com/fare/asdf/blob/master/doc/best_practices.md

---------------

A couple of other things since I did not see your contact information in the repository and this is my only way of contacting you:

1. Johan Andersson has what seems to be a legitimate bugfix at https://github.com/nja/check-it/commit/75a6090bdb6e6f7c0b3c027c1a5125ddd3446e9f Not sure why he did not make a pull request for that.

2. Can you change the license to LGPL version 3 or later? I wrote up a summary of why LLGPL is not necessary here: https://cliki.net/LLGPL I'll add the proper license files, copyright notices, SPDX metadata, etc. if you agree.

DalekBaldwin commented:
Thanks. I'm just now getting some time to work on projects outside of my job again. I updated the CI script to accommodate the system name change.

The only part of the LGPL that always seemed questionable for a macro-heavy Lisp library is the limitation to incorporating "small macros, inline functions and templates (ten or fewer lines in length)". If it is only used in a clearly separated test package, I expect the package under test would be unaffected anyway. But it still seems to be a better fit than any other standard license, and the LLGPL's language about function redefinition raises trickier questions about future ideas I had in mind, so I would welcome the change. You can add it to the existing PR if you like.

vsedach commented:
Ok, I changed the license and added the necessary license files and file copyright notices.

Regarding "small macros, inline functions and templates" (in LGPLv3 this is in section 3), that is not a limitation, but the threshold at which the program would have to provide notice that is using the library under the LGPL.
